### ISSUE TITLE
CIECAM02 implementation

### ISSUE THREAD

**Issue by okaneco** [Ogeon/palette#168]
Have you investigated anything with CIECAM02? I'm interested in this space because of a website I use which has listings of [watercolor pigment colorimetry measurements](https://www.handprint.com/HP/WCL/waterr.html#PR264) in CIECAM Jab and it'd be cool to have. I know there are some differences between Jab and Jch and that it's a complicated model.

I'd be interested in helping implement this but didn't want to be a hassle if I got in the way since I've not contributed something of this scale here.

**Comment by Ogeon** [comment]
I haven't looked into it myself, except for a peek at the wiki page just now, but it looks like it may be an interesting addition. Have I understood it right, that it's basically a successor to Lab? I'll have to read some more later...

Feel free to give it a spin! It's probably good to start small, maybe with only one of them to begin with. Just keep in mind that the conversion traits will be replaced soon-ish (see https://github.com/Ogeon/palette/issues/41#issuecomment-520147523).

**Comment by okaneco** [comment]
Yes, it seems like it'll be the last color model for a while. It corrects for some of the distortions in hue and chroma spacing that are still present in Lab, although Lab is probably enough for most use cases. It also takes into context the colors around a sample as well as different conditions for the viewer such as dim, average, and bright environments. There's further explanation and graphs here.
http://handprint.com/HP/WCL/color7.html#CIECAM

I was aware that you were working on overhauling the traits so I didn't want to dive too deep yet. The 1.41 update seems like a welcome improvement and I'm not sure if 2018 edition lifts additional hurdles. I might plug away at this in the background to get more familiar with the model and implementing it. I don't think it'd be too hard to adapt to whatever changes you make but figured it'd be good to get this idea out there for anyone else that may want to try.

**Comment by Ogeon** [comment]
It will probably be fine to start with it anyway. But there will be conflicts of some kind, depending on which one goes in first. :slightly_smiling_face: I don't think I have an awful lot more to do, other than  the reverse conversion (from Palette colors to user colors) and some final polishing, so it may actually finally be done soon.

Thanks for the link! I'll read a bit tomorrow, when my brain is functioning again.

**Comment by Ogeon** [comment]
Seem complex, indeed, but pretty cool. It would be interesting to see if it can fit into the existing system, or if the additional parameters add too much complexity. I suppose its possible to turn the environmental information into a type parameter, similar to the white point in other types, as a common base assumption for each value. The point of doing it with the white point is to avoid mixing up colors meant for different illuminants, so that would have a similar purpose.
